### PR TITLE
Add support for custom RSS content that includes video URL

### PR DESCRIPTION
**Pull request by carrchr**
Regex looks for `videourl:=` wrapped in an HTML comment
I added the `NSString hasPrefix:` check as an optimization

### CODE REVIEW

File: RSSParser/RSSParser.m [hunk 1]
```diff
@@ -210,6 +219,27 @@ -(NSDate *)getDateFromString:(NSString *)string {
...
+    NSError *error = NULL;
+    NSRegularExpression *regex =
+        [NSRegularExpression regularExpressionWithPattern:@"(?:.*<!--\\s*videourl:=)(http.*)\\s*-->"
+                                                  options:NSRegularExpressionCaseInsensitive
+                                                    error:&error];
+    NSTextCheckingResult *match = [regex firstMatchInString:string
+                                         options:0
+                                           range:NSMakeRange(0, string.length)];
+    if (match) {
+        NSRange matchRange = match.range;   // entire captured tect
```
Comment (joshfriend) [inline RSSParser/RSSParser.m:236]: what is a tect?

**Review by joshfriend — APPROVED** [review]

File: RSSParser/RSSParser.m [hunk 2]
```diff
@@ -109,6 +109,15 @@ - (void)parser:(NSXMLParser *)parser didEndElement:(NSString *)elementName names
             }
         } else if ([elementName isEqualToString:@"content:encoded"] || [elementName isEqualToString:@"content"]) {
             [self.currentParsedItem setContent:self.currentParsedText];
+
+            NSString *string = self.currentParsedText;
+            if ([string hasPrefix:@"<p><!--"] || [string hasPrefix:@"<!--"]) {
```
Comment (jkreiser) [inline RSSParser/RSSParser.m:114]: Is there any chance that the string might also be prefixed with white space?

Comment (carrchr) [reply RSSParser/RSSParser.m:114]: Hmmm... I just tested it on Diocesan's Wordpress. Added a couple spaces _before_ the comment string and they do indeed get sent in the parsed `content:encoded` tag. So I guess I should remove that check and just use the Regex?

Comment (jkreiser) [reply RSSParser/RSSParser.m:114]: That, or trim the whitespace

**Pull request merged**